src/routes/catalog.ts

```diff
@@ -112,6 +119,8 @@ const providerCollections = async (
   );
 
   try {
+    if ("provider" in mergedQuery && mergedQuery.provider == ALL_PROVIDER)
```

Comment:
Here we can simplify the condition by directly comparing mergedQuery.provider to ALL_PROVIDER
```suggestion
    if (mergedQuery.provider === ALL_PROVIDER) {
        delete mergedQuery.provider;
  }
```